src/NuGet.Core/NuGet.Build.Tasks.Console/DependencyGraphSpecGenerator.cs

```diff
@@ -0,0 +1,896 @@
+
+                    if (project != null && BuildTasksUtility.DoesProjectSupportRestore(project))
+                    {
+                        dependencyGraphSpec.AddRestore(entryPoint.ProjectFile);
+                    }
+                }
+
+                sw.Stop();
+
+                // TODO: Localized resource
+                MSBuildLogger.LogDebug(string.Format(CultureInfo.CurrentCulture, "Created DependencyGraphSpec in {0:D2}ms.", sw.ElapsedMilliseconds));
+
+                return dependencyGraphSpec;
+            }
+            catch (Exception e)
+            {
+                LoggingQueue.TaskLoggingHelper.LogErrorFromException(e, showStackTrace: true);
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Gets a <see cref="PackageSpec" /> for the specified project.
+        /// </summary>
+        /// <param name="project">An <see cref="IMSBuildProject" /> object that represents the project.</param>
+        /// <param name="allInnerNodes">An <see cref="IReadOnlyDictionary{String,IMSBuildProject}" /> that represents all inner projects by their target framework.</param>
+        /// <returns></returns>
+        private PackageSpec GetPackageSpec(IMSBuildProject project, IReadOnlyDictionary<string, IMSBuildProject> allInnerNodes)
+        {
+            var settings = RestoreSettingsUtils.ReadSettings(
+                project.GetProperty("RestoreSolutionDirectory"),
+                project.GetProperty("RestoreRootConfigDirectory") ?? project.Directory,
+                UriUtility.GetAbsolutePath(project.Directory, project.GetProperty("RestoreConfigFile")),
+                MachineWideSettingsLazy,
+                _settingsLoadContext);
+
+            // Get the target frameworks for the project and the project instance for each framework
+            var projectsByTargetFramework = GetProjectTargetFrameworks(project, allInnerNodes);
+
+            var restoreMetadataAndTargetFrameworkInformation = GetProjectRestoreMetadataAndTargetFrameworkInformation(project, projectsByTargetFramework, settings);
+
+            var packageSpec = new PackageSpec(restoreMetadataAndTargetFrameworkInformation.TargetFrameworkInfos)
+            {
+                FilePath = project.FullPath,
+                Name = restoreMetadataAndTargetFrameworkInformation.RestoreMetadata.ProjectName,
+                RestoreMetadata = restoreMetadataAndTargetFrameworkInformation.RestoreMetadata,
+                RuntimeGraph = new RuntimeGraph(
+                    MSBuildStringUtility.Split($"{project.GetProperty("RuntimeIdentifiers")};{project.GetProperty("RuntimeIdentifier")}")
+                        .Concat(projectsByTargetFramework.Values.SelectMany(i => MSBuildStringUtility.Split($"{i.GetProperty("RuntimeIdentifiers")};{i.GetProperty("RuntimeIdentifier")}")))
+                        .Distinct(StringComparer.Ordinal)
```

Comment:
Runtime identifiers are opaque strings and therefore case-sensitive?

Reply:
Apparently not: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs#L507